### DISCUSSION TITLE
Should `IDynamoDbContext` be a singleton?

### DISCUSSION THREAD

**Discussion by baynezy** [AllocZero/EfficientDynamoDb#253] [General]
I am interested in knowing what the best practice is here.

**Comment by firenero** [comment]
It's recommended to use a single instance of `DynamoDbContext` for everything to get better cache utilization. However, you need to use different instances of context when you need to utilize different `DynamoDbConfig`. The common cases for that would be accessing different AWS regions or having separate table prefixes.

**Reply by baynezy** [reply]
Thanks.
